batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/deployment/BatchEnvironmentService.java

```diff
@@ -61,14 +62,20 @@
     private final ClassLoader classLoader;
     private final JobXmlResolver jobXmlResolver;
     private final String deploymentName;
+    private final EEModuleDescription eeModuleDescription;
     private final boolean legacySecurityPresent;
     private SecurityAwareBatchEnvironment batchEnvironment = null;
     private volatile ControlPoint controlPoint;
 
-    public BatchEnvironmentService(final ClassLoader classLoader, final JobXmlResolver jobXmlResolver, final String deploymentName, final boolean legacySecurityPresent) {
+    public BatchEnvironmentService(final ClassLoader classLoader,
+                                   final JobXmlResolver jobXmlResolver,
+                                   final String deploymentName,
+                                   final EEModuleDescription eeModuleDescription,
+                                   final boolean legacySecurityPresent) {
         this.classLoader = classLoader;
         this.jobXmlResolver = jobXmlResolver;
         this.deploymentName = deploymentName;
+        this.eeModuleDescription = eeModuleDescription;
```

Comment:
I think we should probably just pass the `NamespaceContextSelector` rather than the `EEModuleDescription`. It seems like we shouldn't be holding a reference to that in a service.

Reply:
Changed to pass `NamespaceContextSelector` instead of `EEModuleDescription`.